File: src/TaskOTime/TaskOTime.ViewModel/ViewModels/DialogRequestedEventArgs.vb
Imports System

Namespace ViewModels
    Public NotInheritable Class DialogRequestedEventArgs
        Inherits EventArgs

        Public Sub New(dialog As DialogShellViewModel)
            If dialog Is Nothing Then
                Throw New ArgumentNullException(NameOf(dialog))
            End If

            Me.Dialog = dialog
        End Sub

        Public ReadOnly Property Dialog As DialogShellViewModel
    End Class
End Namespace

File: src/TaskOTime/TaskOTime.ViewModel/ViewModels/DialogRequestedEventArgs.cs
using System;

namespace TaskOTime.ViewModel.ViewModels
{
    public sealed class DialogRequestedEventArgs : EventArgs
    {

        public DialogRequestedEventArgs(DialogShellViewModel dialog)
        {
            if (dialog is null)
            {
                throw new ArgumentNullException(nameof(dialog));
            }

            Dialog = dialog;
        }

        public DialogShellViewModel Dialog { get; private set; }
    }
}